components/Ripple/src/MDCRippleView.m

```diff
@@ -203,14 +206,38 @@ - (void)beginRippleTouchDownAtPoint:(CGPoint)point
 
 - (void)beginRippleTouchUpAnimated:(BOOL)animated
                         completion:(nullable MDCRippleCompletionBlock)completion {
+  // If all ripple animations are already cancelled and removed from the superlayer call the
+  // short circuit and call the completion handler directly.
+  if (self.activeRippleLayer == nil) {
+    if (completion) {
+      completion();
+      return;
+    }
+  }
```

Comment:
I'm a bit nervous about these extra logical paths. This method went from one to three; can we move this new logic into `endRippleAnimated:completion:`?

Reply:
Ah I see now that we can't due to the call being on `self.activeRippleLayer`.

Reply:
I think this would be more clear then:

```objc
  // If all ripple animations are already cancelled and removed from the superlayer call the
  // short circuit and call the completion handler directly.
  if (self.activeRippleLayer == nil) {
    if (completion) {
      completion();
    }
    return;
  }
```

This results in only two logical paths rather than three.

Reply:
Yep totally agree! Changed.